File: src/main/java/org/team1/nbe1_3_team01/global/util/Response.java
package org.team1.nbe1_3_team01.global.util;

import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;

@AllArgsConstructor(access = AccessLevel.PRIVATE)
@NoArgsConstructor(access = AccessLevel.PRIVATE)
@Getter
public final class Response<T> {
    private String msg;
    private T result;

    public static Response<Void> error(String msg){
        return new Response<>(msg, null);
    }

    public static <T> Response<T> success(T result){
        return new Response<>("Success", result);
    }
}

File: src/main/java/org/team1/nbe1_3_team01/global/util/Response.kt
package org.team1.nbe1_3_team01.global.util

data class Response<T>(
    val msg: String?,
    val result: T?
) {
    companion object {
        fun error(msg: String?): Response<Void> {
            return Response(msg, null)
        }

        fun <T> success(result: T): Response<T> {
            return Response("Success", result)
        }
    }
}
